compact_str/src/tests.rs

```diff
@@ -1850,3 +1850,98 @@ fn multiple_nieches_test() {
         core::mem::size_of::<String>()
     );
 }
+
+#[test]
+fn test_is_empty() {
```

Comment:
It might be worth simplifying this test with `<&str>::repeat()`, something like

```rust
assert!(CompactString::new("").is_empty());
assert!(CompactString::from_static_str("").is_empty());

for i in 1..=25 {
  let s = Box::leak("\0".repeat(i).into_boxed_str());
  assert!(!CompactString::new(s).is_empty());
  assert!(!CompactString::from_static_str(s).is_empty());
}
```

Reply:
Never mind - just realized `CompactString::from_static_str()` is a `const fn` - that won't work with `&'static str`s you get by leaking at runtime

Reply:
I simplified the test a bit.